clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -384,10 +388,27 @@ public CompletableFuture<Void> seekToCheckpoint(Set<VeniceChangeCoordinate> chec
     });
   }
 
-  private void pubSubConsumerSeek(PubSubTopicPartition topicPartition, Long offset) {
+  void checkLiveVersion(String topicName) {
+    Store store = storeRepository.getStore(storeName);
+    try {
+      store.getVersionOrThrow(Version.parseVersionFromVersionTopicName(topicName));
+    } catch (StoreVersionNotFoundException ex) {
+      throw new VeniceCoordinateOutOfRangeException("Checkpoint is off retention!  Version has been deprecated...", ex);
+    }
+  }
+
+  private void pubSubConsumerSeek(PubSubTopicPartition topicPartition, Long offset)
+      throws VeniceCoordinateOutOfRangeException {
     // Offset the seek to next operation inside venice pub sub consumer adapter subscription logic.
     long targetOffset = offset == OffsetRecord.LOWEST_OFFSET ? OffsetRecord.LOWEST_OFFSET : offset - 1;
-    pubSubConsumer.subscribe(topicPartition, targetOffset);
+    try {
+      pubSubConsumer.subscribe(topicPartition, targetOffset);
+    } catch (PubSubTopicDoesNotExistException ex) {
+      throw new VeniceCoordinateOutOfRangeException(
+          "Version does not exist! Checkpoint contained version: " + topicPartition.getTopicName()
```

Comment:
Maybe changed to topic partition not exist to include more information?

Reply:
Like what for example?

Reply:
This exception type is meant to be very specific so that the user knows that the offset they passed isn't on retention.  This is basically a '404' type sort of thing, and they should be reactive to it.  I'd be concerned about topic does not exist maybe conflating things.

Reply:
I mean putting topic partition instance into the message instead of `topicPartition.getTopicName()`